beoordelingstool_dockwidget.py

```diff
@@ -57,10 +130,722 @@ def add_herstelmaatregelen(self):
+            self.tablewidget_pipes.setItem(0, 27, QTableWidgetItem(f["ABS"]))
+            self.tablewidget_pipes.setItem(0, 28, QTableWidgetItem(f["ACA"]))
+            self.tablewidget_pipes.setItem(0, 29, QTableWidgetItem(f["ACB"]))
+            self.tablewidget_pipes.setItem(0, 30, QTableWidgetItem(f["ACC"]))
+            self.tablewidget_pipes.setItem(0, 31, QTableWidgetItem(f["ACD"]))
+            self.tablewidget_pipes.setItem(0, 32, QTableWidgetItem(f["ACG"]))
+            self.tablewidget_pipes.setItem(0, 33, QTableWidgetItem(f["ACJ"]))
+            self.tablewidget_pipes.setItem(0, 34, QTableWidgetItem(f["ACK"]))
+            self.tablewidget_pipes.setItem(0, 35, QTableWidgetItem(f["ACM"]))
+            self.tablewidget_pipes.setItem(0, 36, QTableWidgetItem(f["ACN"]))
+            self.tablewidget_pipes.setItem(0, 37, QTableWidgetItem(f["ADA"]))
+            self.tablewidget_pipes.setItem(0, 38, QTableWidgetItem(f["ADB"]))
+            self.tablewidget_pipes.setItem(0, 39, QTableWidgetItem(f["ADC"]))
+            self.tablewidget_pipes.setItem(0, 40, QTableWidgetItem(f["AXA"]))
+            self.tablewidget_pipes.setItem(0, 41, QTableWidgetItem(f["AXB"]))
+            self.tablewidget_pipes.setItem(0, 42, QTableWidgetItem(f["AXF"]))
+            self.tablewidget_pipes.setItem(0, 43, QTableWidgetItem(f["AXG"]))
+            self.tablewidget_pipes.setItem(0, 44, QTableWidgetItem(f["AXH"]))
+            self.selected_pipe_id = f.id()
+
+    def save_beoordeling_leidingen(self):
+        """Save herstelmaatregel and opmerking in the shapefile."""
+        layer = iface.activeLayer()
+        fid = self.selected_pipe_id
+        herstelmaatregel = str(self.field_combobox_pipes.currentText())
+        opmerking = str(self.value_plaintextedit_pipes.toPlainText())
+        layer.startEditing()
+        layer.changeAttributeValue(fid, 46, herstelmaatregel)  # Herstelmaatregel
+        layer.changeAttributeValue(fid, 47, opmerking)  # Opmerking
+        layer.commitChanges()
+        layer.triggerRepaint()
+
+    def show_measuring_point(self):
+        """Show the measuring station that belongs to a certain pipe."""
+        try:
+            current_measuring_point_pipe_id = self.tablewidget_measuring_points.itemAt(0,0).text()
+        except:  # No measuring point selected
+            current_measuring_point_pipe_id = -1
+        # if current_measuring_point_pipe_id != self.selected_pipe_id:
+        layerList = QgsMapLayerRegistry.instance().mapLayersByName(SHP_NAME_MEASURING_POINTS)
+        if layerList:
+            layer = layerList[0]
+            expr = QgsExpression("\"PIPE_ID\" IS '{}'".format(self.selected_pipe_id))
+            measuring_points = layer.getFeatures(QgsFeatureRequest(expr))  #.next()
+            ids = [measuring_point.id() for measuring_point in measuring_points]  # select only the features for which the expression is true
+            first_id = ids[0]
+            last_id = ids[-1] if ids[-1] else ids[0]
```

Comment:
Deze gaat niet werken :-) Als er geen laatste item is is er ook geen eerste item.
Of beter gezegd: je doet te goed je best.